test-harness/index.ts

```diff
@@ -1,105 +1,72 @@
-import { spawnSync } from 'child_process';
+import * as path from '@stoplight/path';
+import { Dictionary } from '@stoplight/types';
+import * as fg from 'fast-glob';
 import * as fs from 'fs';
-// @ts-ignore
-import * as globFs from 'glob-fs';
-import * as os from 'os';
-import * as path from 'path';
-// @ts-ignore
 import * as tmp from 'tmp';
-import { parseScenarioFile } from './helpers';
-
-const glob = globFs({ gitignore: true });
+import { promisify } from 'util';
+import { applyReplacements, parseScenarioFile, tmpFile } from './helpers';
+import { spawnNode } from './spawn';
+const writeFileAsync = promisify(fs.writeFile);
 
 const spectralBin = path.join(__dirname, '../binaries/spectral');
-
-type Replacement = {
-  from: RegExp;
-  to: string;
-};
-
-function replaceVars(string: string, replacements: Replacement[]) {
-  return replacements.reduce((str, replace) => str.replace(replace.from, replace.to), string);
-}
+const cwd = path.join(__dirname, './scenarios');
+const files = process.env.TESTS ? String(process.env.TESTS).split(',') : fg.sync('**/*.scenario', { cwd });
 
 describe('cli acceptance tests', () => {
-  const cwd = path.join(__dirname, './scenarios');
-  const files = process.env.TESTS ? String(process.env.TESTS).split(',') : glob.readdirSync('**/*.scenario', { cwd });
-
-  files.forEach((file: string) => {
+  describe.each(files)('%s file', file => {
     const data = fs.readFileSync(path.join(cwd, file), { encoding: 'utf8' });
     const scenario = parseScenarioFile(data);
-    const replacements: Replacement[] = [];
-
-    let tmpFileHandle: tmp.FileSyncObject;
-
-    beforeAll(() => {
-      if (scenario.document) {
-        tmpFileHandle = tmp.fileSync({
-          postfix: '.yml',
-          dir: undefined,
-          name: undefined,
-          prefix: undefined,
-          tries: 10,
-          template: undefined,
-          unsafeCleanup: undefined,
-        });
-
-        replacements.push({
-          from: /\{document\}/g,
-          to: tmpFileHandle.name,
-        });
-
-        replacements.push({
-          from: /\{documentWithoutExt\}/g,
-          to: tmpFileHandle.name.replace(/\.yml$/, ''),
-        });
-
-        fs.writeFileSync(tmpFileHandle.name, scenario.document, { encoding: 'utf8' });
-      }
+    const replacements: Dictionary<string> = {
+      __dirname,
+      spectral: spectralBin,
```

Comment:
Keeping in mind the end goal is to open source this, should we maybe instead of "spectral" call it "bin" or something?

Reply:
Yeah, makes sense, I'll do it.